libcudacxx/test/libcudacxx/std/containers/sequences/inplace_vector/assign.pass.cpp

```diff
@@ -0,0 +1,353 @@
+  }
+
+  {
+    const cuda::std::array<T, 2> expected = {T(42), T(42)};
+    cuda::std::inplace_vector<T, 42> input_range_shrink{T(1), T(42), T(1337), T(0)};
+    input_range_shrink.assign_range(Range<T, 2>{T(42), T(42)});
+    assert(!input_range_shrink.empty());
+    assert(cuda::std::equal(input_range_shrink.begin(), input_range_shrink.end(), expected.begin(), expected.end()));
+  }
+
+  {
+    const cuda::std::array<T, 6> expected = {T(42), T(42), T(42), T(42), T(42), T(42)};
+    cuda::std::inplace_vector<T, 42> input_range_grow{T(1), T(42), T(1337), T(0)};
+    input_range_grow.assign_range(Range<T, 6>{T(42), T(42), T(42), T(42), T(42), T(42)});
+    assert(!input_range_grow.empty());
+    assert(cuda::std::equal(input_range_grow.begin(), input_range_grow.end(), expected.begin(), expected.end()));
+  }
+}
+#endif // TEST_STD_VER >= 2017 && !defined(TEST_COMPILER_MSVC_2017)
+
+template <class T>
+__host__ __device__ constexpr void test()
+{
+  {
+    cuda::std::inplace_vector<T, 0> no_capacity_size_value{};
+    no_capacity_size_value.assign(0, T(42));
+    assert(no_capacity_size_value.empty());
+  }
+
+  {
+    using iter = cpp17_input_iterator<const T*>;
+    const cuda::std::initializer_list<T> input{};
+    cuda::std::inplace_vector<T, 0> no_capacity_input_iter_iter{};
+    no_capacity_input_iter_iter.assign(iter{input.begin()}, iter{input.end()});
+    assert(no_capacity_input_iter_iter.empty());
+  }
+
+  {
+    const cuda::std::initializer_list<T> input{};
+    cuda::std::inplace_vector<T, 0> no_capacity_iter_iter{};
+    no_capacity_iter_iter.assign(input.begin(), input.end());
+    assert(no_capacity_iter_iter.empty());
+  }
+
+  {
+    const cuda::std::initializer_list<T> input{};
+    cuda::std::inplace_vector<T, 0> no_capacity_init_list{};
+    no_capacity_init_list.assign(input);
+    assert(no_capacity_init_list.empty());
+  }
+
+  {
+    cuda::std::inplace_vector<T, 42> size_value_empty{};
+    size_value_empty.assign(0, T(42));
+    assert(size_value_empty.empty());
+  }
+
+  {
+    cuda::std::inplace_vector<T, 42> size_value_shrink_empty{T(1), T(42), T(1337), T(0)};
+    size_value_shrink_empty.assign(0, T(42));
+    assert(size_value_shrink_empty.empty());
+  }
+
+  {
+    const cuda::std::array<T, 2> expected = {T(42), T(42)};
+    cuda::std::inplace_vector<T, 42> size_value_shrink{T(1), T(42), T(1337), T(0)};
+    size_value_shrink.assign(2, T(42));
+    assert(!size_value_shrink.empty());
+    assert(cuda::std::equal(size_value_shrink.begin(), size_value_shrink.end(), expected.begin(), expected.end()));
+  }
+
+  {
+    const cuda::std::array<T, 6> expected = {T(42), T(42), T(42), T(42), T(42), T(42)};
+    cuda::std::inplace_vector<T, 42> size_value_grow{T(1), T(42), T(1337), T(0)};
+    size_value_grow.assign(6, T(42));
+    assert(!size_value_grow.empty());
+    assert(cuda::std::equal(size_value_grow.begin(), size_value_grow.end(), expected.begin(), expected.end()));
+  }
+  {
```

Comment:
You added an empty line in other places well.
```suggestion
  }
  
  {
```